tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/IR/hlo_ops.cc

```diff
@@ -2140,9 +2140,55 @@ OpFoldResult ConvertOp::fold(ArrayRef<Attribute> operands) {
   return {};
 }
 
+namespace {
+
+struct EliminateRedundantConvert : public OpRewritePattern<ConvertOp> {
+  using OpRewritePattern<ConvertOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(ConvertOp op,
+                                PatternRewriter &rewriter) const override {
+    if (auto convert_op = op.operand().getDefiningOp<ConvertOp>()) {
```

Comment:
Nit: please use early exit:
```suggestion
    auto convert_op = op.operand().getDefiningOp<ConvertOp>();
    if (!convert_op) return failure();
```

Reply:
done